File: flib/src/main/java/com/f1reking/flib/base/BaseLibActivity.java
package com.f1reking.flib.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author F1ReKing
 * @date 2019/10/16 13:36
 * @Description
 */
public abstract class BaseLibActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        onPreView();
        setContentView(getLayoutId());
        initView();
    }

    abstract void onPreView();

    abstract int getLayoutId();

    abstract void initView();
}

File: flib/src/main/java/com/f1reking/flib/base/BaseLibActivity.kt
package com.f1reking.flib.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * @author: F1ReKing
 * @date: 2019/3/28 22:31
 * @desc:
 */
abstract class BaseLibActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        onPreView()
        setContentView(getLayoutId())
        initView()
    }

    abstract fun onPreView()

    abstract fun getLayoutId(): Int

    abstract fun initView()


}